core/src/main/java/org/opensearch/sql/calcite/CalcitePlanContext.java

```diff
@@ -36,6 +36,8 @@ public class CalcitePlanContext {
   public final QueryType queryType;
   public final Integer querySizeLimit;
 
+  public static final ThreadLocal<Boolean> skipEncoding = ThreadLocal.withInitial(() -> false);
```

Comment:
Not sure if possible to avoid this global variable. If this is necessary, could you add clear javadoc for its purpose?